pkg/network/ifaceutil/interface.go

```diff
@@ -133,25 +134,60 @@ func RandomName(prefix string) (string, error) {
 	return fmt.Sprintf("%s%x", prefix, b), nil
 }
 
-// MakeVethPair creates a veth pair
-func MakeVethPair(name, master string, mtu int) (netlink.Link, error) {
+// MakeVethPair creates a veth pair and assigns it to a network namespace if provided
+// also sets up the peer link to the specified master link
+//
+// - name: name of the veth interface
+// - master: name of the master interface to attach the veth peer to
+// - mtu: mtu
+// - peerPrefix: prefix for the peer veth interface name (if empty, defaults to 'p')
+// - netNs: network namespace to move the veth interface to. (could be nil)
+//
+// Returns the created netlink.Link or an error
+func MakeVethPair(name, master string, mtu int, netNs ns.NetNS) (netlink.Link, error) {
 	masterLink, err := netlink.LinkByName(master)
+	if err != nil {
+		return nil, fmt.Errorf("master link: %s not found: %v", master, err)
+	}
+
+	nsName := filepath.Base(netNs.Path())
```

Comment:
What if netNs is `nil` I see you already calling the function with `nil` as a last argument already in some places